io.qt.qtwebengine.BaseApp.json

```diff
@@ -69,9 +83,14 @@
             ],
             "sources": [
                 {
-                    "type": "archive",
-                    "url": "http://download.qt.io/archive/qt/5.15/5.15.2/submodules/qtwebengine-everywhere-src-5.15.2.tar.xz",
-                    "sha256": "c8afca0e43d84f7bd595436fbe4d13a5bbdb81ec5104d605085d07545b6f91e0"
+                    "type": "git",
+                    "url": "https://code.qt.io/qt/qtwebengine.git",
+                    "tag": "v5.15.5-lts",
+                    "commit": "9711f64c5082040cb76f6da5ef4a16037dbda08f",
+                    "x-checker-data": {
+                        "type": "git",
```

Comment:
Please note that the git type checker currently suffers from flathub/flatpak-external-data-checker#154, it'll cause f-e-d-c to open another PR the next day if the current one wasn't merged.

Reply:
Well, that would be very wasteful with a such large package, I will change this.

Reply:
I think you can either keep using tarballs and apply html type checker to it, or use the github git mirror with json checker to query github API.

Reply:
I can't use tarballs as there are no tarball releases newer than 5.15.2. Downloading at a tarball archive of a tag from the Github mirror won't help because qtwebkit-chromium will be missing and we need a specific commit so f-e-d-c will not help.  
Using the JSON checker with the Github mirror of the official Qt account won't help because the releases return an empty response, and the response for the list of tags is missing a timestamp.  

It seems like the only way that can work is to use the JSON checker with KDE's Gitlab mirror. The following seems to work but it should have a regex to match only 5.15.x releases, and I haven't tried yet to build from KDE.

```
"x-checker-data": {
    "type": "json",
    "url": "https://invent.kde.org/api/v4/projects/qt%2Fqt%2Fqtwebengine/repository/tags",
    "tag-query": ".[0].name",
    "version-query": "$tag",
    "timestamp-query": ".[0].commit.created_at"
}
```

Reply:
I see. Maybe `map(select(.name | test("^v5.15."))) | first` would do?

Reply:
I've been thinking about something more like this:

```
"x-checker-data": {
    "type": "json",
    "url": "https://invent.kde.org/api/v4/projects/qt%2Fqt%2Fqtwebengine/repository/tags",
    "tag-query": "first(.[].name | match( \"v5.15[\\\\d.]+-lts|v5.15[\\\\d.]+\" ) | .string)",
    "version-query": "$tag",
    "timestamp-query": ".[] | select(.name==$tag) | .commit.created_at"
}

```

Reply:
`match()` returns matching pattern. In this case it'll be `v5.15.5`

Reply:
> match() returns matching pattern. In this case it'll be v5.15.5, while the actual tag is v5.15.5-lts

Right, I noticed this and it's why I put `v5.15[\\\\d.]+-lts` before `v5.15[\\\\d.]+`.